src/transactions/TransactionFrame.cpp

```diff
@@ -1168,11 +1182,11 @@ TransactionFrame::applyOperations(SignatureChecker& signatureChecker,
             outerMeta.pushOperationMetas(std::move(operationMetas));
             outerMeta.pushTxChangesAfter(std::move(changesAfter));
 #ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
-            auto isSmartTx = mOperations.at(0)->isSmartOperation();
-            if ((isSmartTx && mOperations.size() != mEvents.size()) ||
-                (!isSmartTx && !mEvents.empty()))
+            auto isSorobanTx = isSoroban();
+            if ((isSorobanTx && mOperations.size() != mEvents.size()) ||
```

Comment:
Now that we only have 1 op per tx, the events probably need to be updated. Right now, the xdr uses a [vector](https://github.com/stellar/stellar-xdr/blob/5e132b2a6c6c2107b04ed11a00a0040e124b7ee0/Stellar-ledger.x#L444) for each operation, but we can only have one op per tx now. We might need to either split the events up by invocation, or remove the nested vector from xdr. This can be done in a follow up PR.

Reply:
I'd say let's not do any special processing as events are attributed to the contracts and there isn't much value in subdividing them further (like if 2 token transfers happen it doesn't matter much if they happened as 2 separate invocations or as 2 sub-contract calls). 
We probably can remove the vector and call this transaction events.

Reply:
I opened an issue - https://github.com/stellar/stellar-xdr/issues/86